### PR TITLE
Adds admin controller

### PR DESCRIPTION
**Pull request by gabru-md**
Adds `/admin/users` and `/admin/users/*` route to get admin details.
Also adds `is_admin` to user table to create admins to authorize them
inside the application

I will work on adding more stuff :star2: 
cheers!

### CODE REVIEW

**Comment by gabru-md** [comment]
ss_prc_admin

I used the `sqlite3` cli to mark myself as an admin. I've also created a function for the same to create admins in the `User.pm` schema. 
@kyzn how should we mark the admin initially?
also, please review this PR and the README one as well.
Thanks :smiley:

**Comment by kyzn** [comment]
Hey @gabru-md, thanks a lot for your PRs! I will review them as soon as I get a chance.

**Comment by gabru-md** [comment]
sure sir!
:smile:

**Comment by gabru-md** [comment]
review @kyzn ? :sweat_smile:

**Comment by kyzn** [comment]
Will do tomorrow night, once the week is over! Thanks for your patience @gabru-md  🙂

File: lib/PRC/Controller/Admin.pm [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+=cut
+
+
+sub user_data: Path('/admin/users'): Args(1) {
+    my ( $self, $c, $id ) = @_;
+    $c->forward('check_admin_status');
+    my $rs = $c->model('PRCDB::User');
+    if(not defined $id) {
+        $c->log->debug('id not given');
+        $c->response->redirect('/admin/users', 303);
```
Comment (kyzn) [inline lib/PRC/Controller/Admin.pm]: Can we also add an error message that says "User cannot be found."?

File: lib/PRC/Controller/Admin.pm [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+    if($user) {
+        $c->stash({
+            template => 'static/html/admin.html',
+            user => $user,
+            active_tab => 'admin'
+        });
+        $c->detach;
+    }else {
+        $c->log->debug('user not found');
```
Comment (kyzn) [inline lib/PRC/Controller/Admin.pm]: This is very similar to above. Let's try to keep debug logs out of master for now. Maybe make the error message visible to admin, just like above.

File: lib/PRC/Controller/Admin.pm [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        $c->response->redirect('/admin/users', 303);
+        $c->detach;
+    }
+}
+
+
+=head2 repos
```
Comment (kyzn) [inline lib/PRC/Controller/Admin.pm]: Let's go with same naming I commented above.

File: lib/PRC/Controller/Admin.pm [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+
+=head2 repos_data
+
+returns the details about repository with repo_id=$id
+
+=cut
+
+sub repos_data: Path('/admin/repos'): Args(1) {
```
Comment (kyzn) [inline lib/PRC/Controller/Admin.pm]: Let's make this singular as well.

File: lib/PRC/Controller/Admin.pm [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    my ( $self, $c, $id ) = @_;
+    $c->forward('check_admin_status');
+    my $rs = $c->model('PRCDB::Repo');
+    if(not defined $id) {
+        $c->log->debug('id not given');
```
Comment (kyzn) [inline lib/PRC/Controller/Admin.pm]: Same as above.

File: lib/PRC/Controller/Admin.pm [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        $c->response->redirect('/admin/repos', 303);
+        $c->detach;
+    }
+
+    my $repo = $rs->search({
+        repo_id => $id
+    })->first;
+
+    if($repo) {
+        my $repo_owner = $c->model('PRCDB::User')->search({
```
Comment (kyzn) [inline lib/PRC/Controller/Admin.pm]: $repo->owner should already return this, as the relationship is defined in lib/PRC/Schema/Result/Repo.pm

File: lib/PRC/Controller/Admin.pm [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+        $c->stash({
+            template => 'static/html/admin.html',
+            active_tab => 'admin',
+            repo => $repo,
+            repo_owner => $repo_owner
+        });
+        $c->detach;
+    }else {
+        $c->log->debug('repo not found');
```
Comment (kyzn) [inline lib/PRC/Controller/Admin.pm]: same as above

File: lib/PRC/Controller/Auth.pm [hunk 8]
```diff
@@ -91,7 +92,9 @@ sub callback :Path('/callback') :Args(0) {
     # If that didn't work, kick out
     $c->forward('login_error') unless $user;
   }
-
+  if($user->github_login eq PRC::Secrets->init_admin) {
```
Comment (kyzn) [inline lib/PRC/Controller/Auth.pm]: Let's not do the init_admin thing. I'd rather manually assign admin roles for now.

File: lib/PRC/Controller/Root.pm [hunk 9]
```diff
@@ -36,7 +36,7 @@ sub auto :Private {
     alert_info    => delete $c->session->{alert_info},
     alert_warning => delete $c->session->{alert_warning},
     alert_danger  => delete $c->session->{alert_danger},
-    $user ? ( logged_in => 1, github_login => $user->github_login ) : (),
+    $user ? ( logged_in => 1, github_login => $user->github_login, is_admin => $user->is_admin ) : (),
```
Comment (kyzn) [inline lib/PRC/Controller/Root.pm]: I understand you are putting this variable to stash to display new route on navigation bar. Let's not advertise these links for now, even when logged in as an admin.

File: lib/PRC/Schema/Result/User.pm [hunk 10]
```diff
@@ -356,5 +358,19 @@ sub available_repos {
   })->all;
 }
 
+
+=head2 make_admin
```
Comment (kyzn) [inline lib/PRC/Schema/Result/User.pm]: Let's not do the init_admin thing. I'd rather manually assign admin roles for now.

File: lib/PRC/Secrets.pm [hunk 11]
```diff
@@ -37,6 +37,15 @@ sub client_secret {
   return shift->_read('CLIENT_SECRET');
 }
 
+=head2 init_admin
```
Comment (kyzn) [inline lib/PRC/Secrets.pm]: Let's not do the init_admin thing. I'd rather manually assign admin roles for now.

File: root/static/html/admin.html [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+    <div style="margin-top: 20px;">
+        [% IF user -%]
+        <h2>User Details</h2>
+        <br>
+            <table>
+                <th>
+                    [% user.github_login -%]
+                </th>
+                <th>
+                    <img src="https://avatars2.githubusercontent.com/u/[% user.github_id -%]" height=100 width=100 style="border-radius: 50%"/>
```
Comment (kyzn) [inline root/static/html/admin.html]: Please don't load the image.

File: root/static/css/prc.css [hunk 13]
```diff
@@ -25,3 +25,8 @@ body {
 table {
   word-break: break-word; /* Make tables readable on mobile */
 }
+
+td {
```
Comment (kyzn) [inline root/static/css/prc.css]: Mind if I ask why you have added this?

File: root/static/html/admin.html [hunk 14]
```diff
@@ -0,0 +1,198 @@
+<div class="container">
```
Comment (kyzn) [inline root/static/html/admin.html]: I'd rather have separate templates for repo/repos/user/users. If they share piece of code, that can go into a separate file to be included from these 4.

File: root/static/html/wrapper.html [hunk 15]
```diff
@@ -3,10 +3,10 @@
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <title>PullRequest.Club</title>
-  <link rel="stylesheet" href="static/css/bootstrap.min.css" type="text/css" />
-  <link rel="stylesheet" href="static/css/prc.css" type="text/css" />
-  <script src="static/js/jquery-3.2.1.slim.min.js" type="text/javascript"></script>
-  <script src="static/js/bootstrap.min.js" type="text/javascript"></script>
+  <link rel="stylesheet" href="/static/css/bootstrap.min.css" type="text/css" />
```
Comment (kyzn) [inline root/static/html/wrapper.html]: Out of curiosity, was this causing issues?

File: root/static/html/wrapper.html [hunk 16]
```diff
@@ -30,6 +30,18 @@
       <li class="nav-item [% IF active_tab == 'settings' -%]active[% END -%]">
         <a class="nav-link" href="/settings">Settings[% IF active_tab == 'settings' -%] <span class="sr-only">(current)</span>[% END -%]</a>
       </li>
+      [% IF is_admin -%]
```
Comment (kyzn) [inline root/static/html/wrapper.html]: Let's not advertise admin menu for now.

**Comment by kyzn** [comment]
Hey @gabru-md, are you still interested in these PRs?

**Comment by gabru-md** [comment]
yes I'll be updating them.
sorry was busy with my university exams for a long time

**Comment by kyzn** [comment]
No worries -- and good luck with your exams 🎉

**Pull request closed without merging**